test/test_autotokenizer.py

```diff
@@ -28,6 +28,20 @@ def test_llama_tokenizer(self):
         actual_ids = ort_tok([text])[0]
         np.testing.assert_array_equal(ids[0], actual_ids)
 
+    def test_llama_tokenizer_id64(self):
+        # replace the official model name after the model is not gated anymore
+        tokenizer = AutoTokenizer.from_pretrained("hf-internal-testing/llama-tokenizer")
+        text = "I was born in 92000, and this is falsé."
+        ids = tokenizer.encode(text, return_tensors="np")
+
+        ort_tok = OrtPyFunction.from_model(gen_processing_models(
+            tokenizer,
+            pre_kwargs={"WITH_DEFAULT_INPUTS": True,
+                        "CAST_TOKEN_ID": True})[0])
+        actual_ids = ort_tok([text])[0]
+        self.assertEqual(actual_ids.dtype, np.int64)
```

Comment:
what was the type before casting? are we adding the `CAST_TOKEN_ID` option just for this Llama use case?

i suggest changing the `CAST_TOKEN_ID` option to a string type rather than boolean, and changing the implementation to allow casting to _all_ supported TensorProto types. Then we can feed the string input (ex: "int64") to a dict similar to the one we have for tokenizers in `_hf_cvt.py` and cast to the respective type. That way this option can be used to cast for all supported types.

Otherwise, we may rename the option to something like `CAST_TO_INT64` for now.

Reply:
it's not LLAMA only, right now, some tokenizer generated int64 token-id, other did it with int32. But make the data type changes on the existing tokenizer output may cause some unexpected compatible issue. So this option was added is to give user a chance to have a consistent experience.

The full name should be CAST_TOKEN_ID_TO_INT64, to be short, CAST_TOKEN_ID was chose to let the user know only token-id was casted, not any outputs.